### PR TITLE
Security: fix an issue where existing secrets were overwritten

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <grant@portworx.com>

### CODE REVIEW

File: drivers/storage/portworx/components_test.go [hunk 1]
```diff
@@ -3078,6 +3078,33 @@ func TestSecurityInstall(t *testing.T) {
...
+	err = k8sClient.Create(context.TODO(), sharedSecret)
+	require.NoError(t, err)
+	cluster.Spec.Security.Auth.SelfSigned.SharedSecret = &existingSharedSecretName
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	err = testutil.Get(k8sClient, sharedSecret, existingSharedSecretName, cluster.Namespace)
+	require.NoError(t, err)
+	s1, ok := sharedSecret.Data["key1"]
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/components_test.go:3102]: Should we check that the actual shared-secret key is added to the secret?

File: drivers/storage/portworx/component/security.go [hunk 2]
```diff
@@ -251,12 +251,12 @@ func (c *security) createPrivateKeysSecret(
 		},
 	}
 
-	err = k8sutil.CreateOrUpdateSecret(c.k8sClient, sharedSecret, nil)
+	err = k8sutil.CreateOrAppendToSecret(c.k8sClient, sharedSecret, nil)
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/component/security.go:254]: Not related to this change, but general question - When user is passing their own system and app secret as env variable, we will just end up creating a `px-system-secrets` secret and not using it, right?

**Review by piyush-nimbalkar — APPROVED** [review]

File: drivers/storage/portworx/components_test.go (hunk 1 above)
Comment (ggriffiths) [reply drivers/storage/portworx/components_test.go:3102]: Added this check.

File: drivers/storage/portworx/component/security.go (hunk 2 above)
Comment (ggriffiths) [reply drivers/storage/portworx/component/security.go:254]: Answered offline - 
This is almost true. We always create and maintain `px-system-secrets`. We directly use the `px-system-secrets` for getting the apps secret for the operator to communicate w/ PX.

**Pull request merged**